spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkMicroBatchStream.java

```diff
@@ -309,6 +312,49 @@ private static StreamingOffset determineStartingOffset(Table table, Long fromTim
     }
   }
 
+  private static int getMaxFiles(ReadLimit readLimit) {
+    if (readLimit instanceof ReadMaxFiles) {
+      return ((ReadMaxFiles) readLimit).maxFiles();
+    }
+
+    if (readLimit instanceof CompositeReadLimit) {
+      // We do not expect a CompositeReadLimit to contain a nested CompositeReadLimit.
+      // In fact, it should only be a composite of two or more of ReadMinRows, ReadMaxRows and
+      // ReadMaxFiles, with no more than one of each.
+      ReadLimit[] limits = ((CompositeReadLimit) readLimit).getReadLimits();
+      for (int i = 0; i < limits.length; i++) {
+        ReadLimit limit = limits[i];
+        if (limit instanceof ReadMaxFiles) {
+          return ((ReadMaxFiles) limit).maxFiles();
+        }
+      }
```

Comment:
[minor] can we use this ? 
```suggestion
      for (ReadLimit limit: limits) {
        if (limit instanceof ReadMaxFiles) {
          return ((ReadMaxFiles) limit).maxFiles();
        }
      }
```

Reply:
Adopted.